kolibri/plugins/management/assets/src/vue/user-page/index.vue

```diff
@@ -145,30 +144,25 @@
         // creates array of words in filter, removes empty strings
         const searchFilter = this.searchFilter.split(' ').filter(Boolean).map(
           // returns an array of search parameters, ignoring case
-          (query) => new RegExp(query, 'i'));
+          (query) => {
+            const escapedQuery = query.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+            return new RegExp(escapedQuery, 'i');
+          });
 
         return this.users.filter((user) => {
           // fullname created using es6 templates
           const names = [user.full_name, user.username];
 
-          let hasRole = true;
+          let isKind = true;
           let hasName = true;
```

Comment:
Since I mentioned functional vs imperative programming before:

Within a `filter` method (and `map` also), it's helpful to conceptualize the inner function's main purpose as operating on a single item. It can be good to avoid thinking about 'iteration' with these.

The result is that parts of the logic can be cleanly decomposed. For example, something like:

```javascript
methods: {
  userComparator(user1, user2) {
    if (user1.username[0] > user2.username[0]) { return 1; }
    else if (user1.username[0] < user2.username[0]) { return -1; }
    return 0;
  },
  matchesTextFilter(user) {
    const full_name = user.full_name.toLowerCase();
    const username = user.username.toLowerCase();
    const terms = this.searchFilter.split(' ').filter(Boolean).map(val => val.toLowerCase());
    return terms.every(term => full_name.includes(term) || username.includes(term));
  },
  matchesRoleFilter(user) {
    if (this.roleFilter === 'all') {
      return true;
    }
    return user.kind === this.roleFilter;
  },
}
computed: {
  visibleUsers() {
    return this.users
      .filter(user => this.matchesTextFilter(user) && this.matchesRoleFilter(user))
      .sort(this.userComparator);
  },
},
```